charts/redpanda/values.go

```diff
@@ -265,7 +360,79 @@ type Storage struct {
 	} `json:"tieredStoragePersistentVolume" jsonschema:"deprecated"`
 }
 
-func (Storage) JSONSchemaExtend(schema *jsonschema.Schema) {
+func (s *Storage) Translate() map[string]any {
+	result := map[string]any{}
+
+	if !s.IsTieredStorageEnabled() {
+		return result
+	}
+
+	tieredStorageConfig := s.GetTieredStorageConfig()
+	for k, v := range tieredStorageConfig {
+		if v == nil || helmette.Empty(v) {
+			continue
+		}
+
+		// cloud_storage_cache_size can be represented as Resource.Quantity that why value can be converted
+		// from value with SI suffix to bytes number.
+		if asStr, isStr := v.(string); k == "cloud_storage_cache_size" && isStr && asStr != "" {
+			result[k] = helmette.ToJSON(SIToBytes(v.(string)))
+			continue
+		}
+		if k == "cloud_storage_cache_size" {
+			if str, isStr := v.(string); isStr && str != "" {
+				result[k] = helmette.ToJSON(SIToBytes(str))
+			} else if f, isFloat := helmette.AsNumeric(v); isFloat {
+				result[k] = helmette.ToJSON(SIToBytes(helmette.ToString(int(f))))
+			}
+			continue
+		}
+
+		if str, ok := v.(string); ok {
+			result[k] = str
+		} else if b, ok := v.(bool); ok {
+			result[k] = b
+		} else if f, isFloat := helmette.AsNumeric(v); isFloat {
+			result[k] = int(f)
+		} else {
+			result[k] = helmette.MustToJSON(v)
+		}
+	}
+
+	return result
+}
+
+func (s *Storage) IsTieredStorageEnabled() bool {
+	conf := s.GetTieredStorageConfig()
+
+	b, ok := conf["cloud_storage_enabled"]
+	return ok && b.(bool)
+}
+
+func (s *Storage) GetTieredStorageConfig() TieredStorageConfig {
+	result := s.Tiered.Config
```

Comment:
stylistic nit: when possible prefer to rely on early returns to reduce the overall complexity of a function

This code is functionally equivalent but much more straight forward to read because you're never going "what's the value of result?"
```go
if len(s.TieredConfig) > 0 {
    return s.TieredConfig
}
return s.Tiered.Config
```

Reply:
You are totally right